### PR TITLE
add helper methods for SA token secret checks

### PR DESCRIPTION
**Pull request by deads2k**
Determining if a secret is a service account API token secret or getting a list of SA tokens for a given SA requires deep knowledge of which annotations having to match.

This pull creates helper functions to make that determination easier and keeps the knowledge of how the annotations are shaped contained.

@liggitt @csrwng I think this is what we want downstream to keep it from spreading everywhere.

### CODE REVIEW

File: pkg/controller/serviceaccount/util.go [hunk 1]
```diff
@@ -81,3 +83,40 @@ func UserInfo(namespace, name, uid string) user.Info {
 		Groups: MakeGroupNames(namespace, name),
 	}
 }
+
+// GetServiceAccountTokens returns all ServiceAccountToken secrets for the given ServiceAccount
+func GetServiceAccountTokens(secretsNamespacer client.SecretsNamespacer, sa *api.ServiceAccount) ([]*api.Secret, error) {
```
Comment (liggitt) [inline pkg/controller/serviceaccount/util.go]: You can have tokens that are valid but not mountable.

Comment (deads2k) [inline pkg/controller/serviceaccount/util.go]: > You can have tokens that are valid but not mountable.

fixed

File: plugin/pkg/admission/serviceaccount/admission.go [hunk 2]
```diff
@@ -273,19 +274,11 @@ func (s *serviceAccount) getServiceAccountTokens(serviceAccount *api.ServiceAcco
...
-		name := token.Annotations[api.ServiceAccountNameKey]
-		uid := token.Annotations[api.ServiceAccountUIDKey]
-		if name != serviceAccount.Name {
-			// Name must match
-			continue
-		}
-		if len(uid) > 0 && uid != string(serviceAccount.UID) {
-			// If UID is set, it must match
+
+		if !serviceaccount.IsServiceAccountToken(token, serviceAccount) {
```
Comment (liggitt) [inline plugin/pkg/admission/serviceaccount/admission.go]:
```
if serviceaccount.IsServiceAccountToken(token, serviceAccount) {
   tokens = append(tokens, token)
}
```

**Comment by liggitt** [comment]
nit, lgtm once tests pass

**Comment by deads2k** [comment]
@liggitt green

**Comment by liggitt** [comment]
LGTM

**Pull request merged**